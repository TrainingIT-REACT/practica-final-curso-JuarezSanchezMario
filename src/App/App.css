.App {
  height: 100%;
  width: 100%;
  position: absolute;
}

.AppContainer {
  height: 100%;
  width: 100%;
}
.vistaMenu {
  align-items: center;
  height: 90vh;
  background-color: pink;
  width: 20%;
  display: inline-block;
}

.vistaMenu li {
  padding: 1em;
}
.vistaReproductor {
  border: -2px solid rgb(27, 8, 197);
  width: 100%;
  /* bottom: 10px; */
  /* vertical-align: bottom; */
  background-color: blue;
  /* clear: both; */
  /* width: 80%; */
  height: 10vh;
  /* float: left; */
  display: inline-block;
}
.vistaCentro{
  height: 90vh;
    width: 80%;
    background-color: green;
    display: inline-block;
    vertical-align: top;
}
