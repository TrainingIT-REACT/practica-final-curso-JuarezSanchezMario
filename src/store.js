import { createStore, applyMiddleware, combineReducers } from "redux";
import thunk from "redux-thunk";
import { composeWithDevTools } from 'redux-devtools-extension';


// Reducers
import albums from "./reducers/albums";
import historico from "./reducers/historico";
import songs from "./reducers/songs";
import user from "./reducers/user";

export const initialState = {
  albums: {
    albums: [
      {
        id: 1,
        name: 'Chip off the old block',
        artist: 'Blair',
        cover: '/images/cover.jpg'
      },
      {
        id: 2,
        name: 'Battle grounds',
        artist: 'Florian',
        cover: '/images/cover.jpg'
      },
      {
        id: 3,
        name: 'Decisions decisions',
        artist: 'Skylar',
        cover: '/images/cover.jpg'
      },
      {
        id: 4,
        name: 'Blank canvas',
        artist: 'Skylar',
        cover: '/images/cover.jpg'
      },
      {
        id: 5,
        name: 'Ice cold',
        artist: 'Kimberley',
        cover: '/images/cover.jpg'
      },
      {
        id: 6,
        name: 'Honesty',
        artist: 'Ted Garrett',
        cover: '/images/cover.jpg'
      },
      {
        id: 7,
        name: 'Creative director',
        artist: 'Delaney',
        cover: '/images/cover.jpg'
      },
      {
        id: 8,
        name: 'No guarantees',
        artist: 'Kimberley',
        cover: '/images/cover.jpg'
      },
      {
        id: 9,
        name: 'Don\'t push this button',
        artist: 'Delaney',
        cover: '/images/cover.jpg'
      },
      {
        id: 10,
        name: 'Cherry',
        artist: 'Stacy',
        cover: '/images/cover.jpg'
      }
    ],
    loading: false,
    recomendados: [
      {
        id: 8,
        name: 'No guarantees',
        artist: 'Kimberley',
        cover: '/images/cover.jpg'
      },
      {
        id: 9,
        name: 'Don\'t push this button',
        artist: 'Delaney',
        cover: '/images/cover.jpg'
      },
      {
        id: 1,
        name: 'Chip off the old block',
        artist: 'Blair',
        cover: '/images/cover.jpg'
      },
      {
        id: 7,
        name: 'Creative director',
        artist: 'Delaney',
        cover: '/images/cover.jpg'
      },
      {
        id: 2,
        name: 'Battle grounds',
        artist: 'Florian',
        cover: '/images/cover.jpg'
      }
    ],
    selectedAlbum: {
      id: 5,
      name: 'Ice cold',
      artist: 'Kimberley',
      cover: '/images/cover.jpg'
    }
  },
  historico: {
    albums: [
      {
        id: 1,
        name: 'Chip off the old block',
        artist: 'Blair',
        cover: '/images/cover.jpg'
      },
      {
        id: 5,
        name: 'Ice cold',
        artist: 'Kimberley',
        cover: '/images/cover.jpg'
      }
    ],
    songs: [
      {
        id: 1,
        name: 'Take On Me',
        audio: '/music/funky_energy_loop.mp3',
        seconds: 203,
        album_id: 1
      },
      {
        id: 2,
        name: 'S.O.S.',
        audio: '/music/funky_energy_loop.mp3',
        seconds: 203,
        album_id: 1
      }
    ]
  },
  songs: {
    loading: false,
    songs: [
      {
        id: 21,
        name: 'Back In Black',
        audio: '/music/funky_energy_loop.mp3',
        seconds: 203,
        album_id: 5
      },
      {
        id: 22,
        name: 'Rock and Roll Ain\'t Noise Pollution',
        audio: '/music/funky_energy_loop.mp3',
        seconds: 203,
        album_id: 5
      },
      {
        id: 23,
        name: 'Dirty Deeds Done Dirt Cheap',
        audio: '/music/funky_energy_loop.mp3',
        seconds: 203,
        album_id: 5
      },
      {
        id: 24,
        name: 'Whole Lotta Rosie',
        audio: '/music/funky_energy_loop.mp3',
        seconds: 203,
        album_id: 5
      },
      {
        id: 25,
        name: 'Let There Be Rock',
        audio: '/music/funky_energy_loop.mp3',
        seconds: 203,
        album_id: 5
      }
    ]
  },
  user: {
    name: 'Mario',
    email: 'mjs.mariojuarezsanchez@gmail.com',
    logged: true
  }
}

export default createStore(
  combineReducers({albums, historico, songs, user}),
  composeWithDevTools(applyMiddleware(thunk))
);
