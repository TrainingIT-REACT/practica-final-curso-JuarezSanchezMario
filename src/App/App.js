import React, { Component } from "react";
import { Router, Route } from "react-router-dom";
import history from "./history";
import Home from "../Home/Home";
import Albums from "../Albums/Albums";
import Nav from "react-bootstrap/Nav";
import Navbar from "react-bootstrap/Navbar";
import Container from "react-bootstrap/Container";
import Button from "react-bootstrap/Button";
import ButtonToolbar from "react-bootstrap/ButtonToolbar";

// Css
import "./App.css";

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      loading: true,
      albums: []
    };
  }

  componentDidMount() {

  }

  render() {
    return (
      <div className="App">
        <Router history={history}>
        <div className="AppContainer">
          <div className="vistaMenu">
            <Navbar bg="dark" variant="dark">
              <Container>
                <Navbar.Brand
                  style={{ cursor: "pointer" }}
                  onClick={() => history.push("/")}
                >
                  Spotiphy
                </Navbar.Brand>
                <Nav>
                  <Nav.Link style={{"padding-right":"50px"}} onClick={() => history.push("/")}>Inicio</Nav.Link>
                  <Nav.Link style={{"padding-right":"50px"}} onClick={() => history.push("/albums")}>
                    Albums
                  </Nav.Link>
                  <Nav.Link style={{"padding-right":"50px"}} onClick={() => console.log("asdasd")}>
                    Perfil
                  </Nav.Link>
                </Nav>
                <ButtonToolbar>
                  <Button variant="outline-info" to="/pefil">
                    Inicio Sesi√≥n
                  </Button>
                </ButtonToolbar>
              </Container>
            </Navbar>
            <div className="vistaCentro">
              <Route path="/" exact component={Home} />
              <Route path="/session" component={Home} />
              <Route path="/albums" component={Albums} />
              <Route path="/pefil" component={Albums} />
            </div>
            </div>
          </div>
        </Router>
      </div>
    );
  }
}

export default App;
