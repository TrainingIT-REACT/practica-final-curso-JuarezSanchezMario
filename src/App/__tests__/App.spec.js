import React from "react";
import { mount } from "enzyme";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { initialState } from "../../store";
import thunk from "redux-thunk";


// Componente
import App from '../App';

describe("App", () => {
  const mockStore = configureStore([thunk]);
  let store, wrapper;

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = mount(
      <Provider store={store}>
        <App />
      </Provider>
    );
  });

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});