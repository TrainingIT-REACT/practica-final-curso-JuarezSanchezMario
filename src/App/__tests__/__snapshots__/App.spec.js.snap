// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(App)>
    <App
      deleteUser={[Function]}
      displayFetchAlbums={[Function]}
      user={
        Object {
          "email": "mjs.mariojuarezsanchez@gmail.com",
          "logged": true,
          "name": "Mario",
        }
      }
    >
      <div
        className="App"
      >
        <Suspense
          fallback="loading..."
        >
          <Router
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
          >
            <div
              className="AppContainer"
            >
              <div
                className="vistaMenu"
              >
                <Navbar
                  bg="dark"
                  collapseOnSelect={false}
                  expand={true}
                  variant="dark"
                >
                  <nav
                    className="navbar navbar-expand navbar-dark bg-dark"
                  >
                    <Container
                      fluid={false}
                    >
                      <div
                        className="container"
                      >
                        <NavbarBrand
                          onClick={[Function]}
                          style={
                            Object {
                              "color": "#ffefefd1",
                              "cursor": "pointer",
                            }
                          }
                        >
                          <span
                            className="navbar-brand"
                            onClick={[Function]}
                            style={
                              Object {
                                "color": "#ffefefd1",
                                "cursor": "pointer",
                              }
                            }
                          >
                            Spotiphy
                          </span>
                        </NavbarBrand>
                        <Nav
                          fill={false}
                          justify={false}
                        >
                          <ForwardRef
                            as="div"
                            className="navbar-nav"
                            onSelect={[Function]}
                          >
                            <div
                              className="navbar-nav"
                              onKeyDown={[Function]}
                            >
                              <NavLink
                                as={
                                  Object {
                                    "$$typeof": Symbol(react.forward_ref),
                                    "displayName": "SafeAnchor",
                                    "render": [Function],
                                  }
                                }
                                disabled={false}
                                onClick={[Function]}
                                style={
                                  Object {
                                    "color": "#ffefefd1",
                                    "paddingRight": "50px",
                                  }
                                }
                              >
                                <ForwardRef
                                  as={
                                    Object {
                                      "$$typeof": Symbol(react.forward_ref),
                                      "displayName": "SafeAnchor",
                                      "render": [Function],
                                    }
                                  }
                                  className="nav-link"
                                  disabled={false}
                                  onClick={[Function]}
                                  style={
                                    Object {
                                      "color": "#ffefefd1",
                                      "paddingRight": "50px",
                                    }
                                  }
                                >
                                  <SafeAnchor
                                    className="nav-link"
                                    data-rb-event-key={null}
                                    disabled={false}
                                    onClick={[Function]}
                                    style={
                                      Object {
                                        "color": "#ffefefd1",
                                        "paddingRight": "50px",
                                      }
                                    }
                                  >
                                    <a
                                      className="nav-link"
                                      data-rb-event-key={null}
                                      href="#"
                                      onClick={[Function]}
                                      onKeyDown={[Function]}
                                      role="button"
                                      style={
                                        Object {
                                          "color": "#ffefefd1",
                                          "paddingRight": "50px",
                                        }
                                      }
                                    >
                                      Inicio
                                    </a>
                                  </SafeAnchor>
                                </ForwardRef>
                              </NavLink>
                              <NavLink
                                as={
                                  Object {
                                    "$$typeof": Symbol(react.forward_ref),
                                    "displayName": "SafeAnchor",
                                    "render": [Function],
                                  }
                                }
                                disabled={false}
                                onClick={[Function]}
                                style={
                                  Object {
                                    "color": "#ffefefd1",
                                    "paddingRight": "50px",
                                  }
                                }
                              >
                                <ForwardRef
                                  as={
                                    Object {
                                      "$$typeof": Symbol(react.forward_ref),
                                      "displayName": "SafeAnchor",
                                      "render": [Function],
                                    }
                                  }
                                  className="nav-link"
                                  disabled={false}
                                  onClick={[Function]}
                                  style={
                                    Object {
                                      "color": "#ffefefd1",
                                      "paddingRight": "50px",
                                    }
                                  }
                                >
                                  <SafeAnchor
                                    className="nav-link"
                                    data-rb-event-key={null}
                                    disabled={false}
                                    onClick={[Function]}
                                    style={
                                      Object {
                                        "color": "#ffefefd1",
                                        "paddingRight": "50px",
                                      }
                                    }
                                  >
                                    <a
                                      className="nav-link"
                                      data-rb-event-key={null}
                                      href="#"
                                      onClick={[Function]}
                                      onKeyDown={[Function]}
                                      role="button"
                                      style={
                                        Object {
                                          "color": "#ffefefd1",
                                          "paddingRight": "50px",
                                        }
                                      }
                                    >
                                      Albums
                                    </a>
                                  </SafeAnchor>
                                </ForwardRef>
                              </NavLink>
                              <NavLink
                                as={
                                  Object {
                                    "$$typeof": Symbol(react.forward_ref),
                                    "displayName": "SafeAnchor",
                                    "render": [Function],
                                  }
                                }
                                disabled={false}
                                onClick={[Function]}
                                style={
                                  Object {
                                    "color": "#ffefefd1",
                                    "paddingRight": "50px",
                                  }
                                }
                              >
                                <ForwardRef
                                  as={
                                    Object {
                                      "$$typeof": Symbol(react.forward_ref),
                                      "displayName": "SafeAnchor",
                                      "render": [Function],
                                    }
                                  }
                                  className="nav-link"
                                  disabled={false}
                                  onClick={[Function]}
                                  style={
                                    Object {
                                      "color": "#ffefefd1",
                                      "paddingRight": "50px",
                                    }
                                  }
                                >
                                  <SafeAnchor
                                    className="nav-link"
                                    data-rb-event-key={null}
                                    disabled={false}
                                    onClick={[Function]}
                                    style={
                                      Object {
                                        "color": "#ffefefd1",
                                        "paddingRight": "50px",
                                      }
                                    }
                                  >
                                    <a
                                      className="nav-link"
                                      data-rb-event-key={null}
                                      href="#"
                                      onClick={[Function]}
                                      onKeyDown={[Function]}
                                      role="button"
                                      style={
                                        Object {
                                          "color": "#ffefefd1",
                                          "paddingRight": "50px",
                                        }
                                      }
                                    >
                                      Perfil
                                    </a>
                                  </SafeAnchor>
                                </ForwardRef>
                              </NavLink>
                            </div>
                          </ForwardRef>
                        </Nav>
                        <ButtonToolbar
                          role="toolbar"
                        >
                          <div
                            className="btn-toolbar"
                            role="toolbar"
                          >
                            <Button
                              active={false}
                              disabled={false}
                              onClick={[Function]}
                              type="button"
                              variant="info"
                            >
                              <button
                                className="btn btn-info"
                                disabled={false}
                                onClick={[Function]}
                                type="button"
                              >
                                Cerrar Sesión
                              </button>
                            </Button>
                          </div>
                        </ButtonToolbar>
                      </div>
                    </Container>
                    <Badge
                      onClick={[Function]}
                      pill={true}
                      style={
                        Object {
                          "cursor": "pointer",
                        }
                      }
                      variant="primary"
                    >
                      <span
                        className="badge badge-pill badge-primary"
                        onClick={[Function]}
                        style={
                          Object {
                            "cursor": "pointer",
                          }
                        }
                      >
                        Mario
                         
                        mjs.mariojuarezsanchez@gmail.com
                      </span>
                    </Badge>
                  </nav>
                </Navbar>
                <div
                  className="vistaCentro"
                >
                  <Suspense
                    fallback="loading Home..."
                  >
                    <Route
                      component={
                        Object {
                          "$$typeof": Symbol(react.lazy),
                          "_ctor": [Function],
                          "_result": Object {
                            "$$typeof": Symbol(react.memo),
                            "WrappedComponent": [Function],
                            "compare": null,
                            "displayName": "Connect(Home)",
                            "type": [Function],
                            Symbol(Symbol.hasInstance): [Function],
                          },
                          "_status": 1,
                        }
                      }
                      exact={true}
                      path="/"
                    />
                    loading Home...
                  </Suspense>
                  <Suspense
                    fallback="loading Session..."
                  >
                    <Route
                      component={
                        Object {
                          "$$typeof": Symbol(react.lazy),
                          "_ctor": [Function],
                          "_result": null,
                          "_status": -1,
                        }
                      }
                      path="/session"
                    />
                  </Suspense>
                  <Suspense
                    fallback="loading Albums..."
                  >
                    <Route
                      component={
                        Object {
                          "$$typeof": Symbol(react.lazy),
                          "_ctor": [Function],
                          "_result": null,
                          "_status": -1,
                        }
                      }
                      path="/albums"
                    />
                  </Suspense>
                  <Suspense
                    fallback="loading Album..."
                  >
                    <Route
                      component={
                        Object {
                          "$$typeof": Symbol(react.lazy),
                          "_ctor": [Function],
                          "_result": null,
                          "_status": -1,
                        }
                      }
                      path="/album"
                    />
                  </Suspense>
                  <Suspense
                    fallback="loading Perfil..."
                  >
                    <Route
                      component={
                        Object {
                          "$$typeof": Symbol(react.lazy),
                          "_ctor": [Function],
                          "_result": null,
                          "_status": -1,
                        }
                      }
                      path="/perfil"
                    />
                  </Suspense>
                </div>
              </div>
            </div>
          </Router>
        </Suspense>
      </div>
    </App>
  </Connect(App)>
</Provider>
`;
